api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -106,47 +111,59 @@ public Cohort deleteCohort(Cohort cohort) throws DAOException {
 		sessionFactory.getCurrentSession().delete(cohort);
 		return null;
 	}
-	
+
 	/**
 	 * @see org.openmrs.api.db.CohortDAO#getCohorts(java.lang.String)
 	 */
 	@Override
-	@SuppressWarnings("unchecked")
 	public List<Cohort> getCohorts(String nameFragment) throws DAOException {
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Cohort.class);
-		criteria.add(Restrictions.ilike("name", nameFragment, MatchMode.ANYWHERE));
-		criteria.addOrder(Order.asc("name"));
-		return criteria.list();
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<Cohort> query = cb.createQuery(Cohort.class);
+		Root<Cohort> root = query.from(Cohort.class);
+
+		query.where(cb.like(cb.lower(root.get("name")), 
+			MatchMode.ANYWHERE.toCaseInsensitivePattern(nameFragment)));
+		query.orderBy(cb.asc(root.get("name")));
+
+		return session.createQuery(query).getResultList();
 	}
-	
+
 	/**
 	 * @see org.openmrs.api.db.CohortDAO#getAllCohorts(boolean)
 	 */
 	@Override
-	@SuppressWarnings("unchecked")
 	public List<Cohort> getAllCohorts(boolean includeVoided) throws DAOException {
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Cohort.class);
-		
-		criteria.addOrder(Order.asc("name"));
-		
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<Cohort> query = cb.createQuery(Cohort.class);
+		Root<Cohort> root = query.from(Cohort.class);
+
+		List<Predicate> predicates = new ArrayList<>();
+
 		if (!includeVoided) {
-			criteria.add(Restrictions.eq(VOIDED, false));
+			predicates.add(cb.isFalse(root.get(VOIDED)));
 		}
-		
-		return (List<Cohort>) criteria.list();
+
+		query.where(predicates.toArray(new Predicate[0]));
```

Comment:
Wouldn't it be enough to just do? `query.where(cb.isFalse(root.get(VOIDED)))`

Reply:
Yes that's more concise, updated.